### PR TITLE
Exersices 1

### PR DESCRIPTION
**Pull request by Andkleven**
Have done every task on exercise 1

### CODE REVIEW

File: src/anders_mathiesen_ex/ex01/comp_to_loop.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+def squares_by_loop(n):
+    squares_by_list = []
+    for k in range(n):
+        if k % 3 == 1:
+            squares_by_list.append(k**2)
+    return squares_by_list
+
+
+print(squares_by_comp(5))
+print(squares_by_loop(5))
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/comp_to_loop.py]: Avoid print statements outside the main section of the program.

File: src/anders_mathiesen_ex/ex01/deck_by_comp.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return deck
+
+
+def deck_comp():
+    return [(suit, val) for suit in SUITS for val in VALUES]
+
+
+if __name__ == '__main__':
+    if deck_loop() != deck_comp():
+        print('ERROR!')
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/deck_by_comp.py]: no blank line at end of file (PEP8 violation)

File: src/anders_mathiesen_ex/ex01/letter_counts.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    return frequencies
+
+
+if __name__ == '__main__':
+    text = input('Please enter text to analyse: ')
+
+    frequencies = letter_freq(text)
+    for letter, count in frequencies.items():
+        print('{:3}{:10}'.format(letter, count))
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/letter_counts.py]: no blank line at end of file (PEP8 violation)

File: src/anders_mathiesen_ex/ex01/letter_counts.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+
+def letter_freq(txt):
+    frequencies = {}
+    for symbol in list(txt):
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/letter_counts.py:6]: this is not case-insensitive, i.e. 
`In [2]: letter_freq("HHhhhh")                                                      
Out[2]: {'H': 2, 'h': 4}`

File: src/anders_mathiesen_ex/ex01/tidy_code.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+from random import randint
+
+__author__ = 'Anders Mathiesen'
+__email__ = 'andermat@nmbu.no'
+
+
+def get_guess_bigger_then_or_lik_1():
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/tidy_code.py:9]: please write all code in english. Also i think this function definition is a bit too explicit. could be something like `user_guess()`

File: src/anders_mathiesen_ex/ex01/tidy_code.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    guess = 0
+    while guess < 1:
+        guess = input('Your guess bigger then or lik 1: ')
+        try:
+            guess = int(guess)
+        except:
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/tidy_code.py]: This is pretty cool, though be wary of using a "bare" except statement as it is usually supposed to catch errors. A better option in this case would be an if-test.

File: src/anders_mathiesen_ex/ex01/tidy_code.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            print('Enter a number')
+            guess = 0
+    return guess
+
+
+def get_random_number():
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/tidy_code.py:21]: roll_dice()?

File: src/anders_mathiesen_ex/ex01/tidy_code.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    return randint(1, 6) + randint(1, 6)
+
+
+def check_if_guess_is_correct(random_number, guess):
```
Comment (danielambrosius) [inline src/anders_mathiesen_ex/ex01/tidy_code.py]: conventions is to write func names that return booleans as a "questions". i.e. just is_correct. In this case though, `is_equal()` would be much better, as that is what the function does irrespective of the rest of the script.

**Review by danielambrosius — CHANGES_REQUESTED** [review]
Code looks all right except the letter frequencies (see comment). And i would try to not have overly explicit function/variable names.

Unfortunately your code fails PEP8 (see below), so i cannot approve it. You should be fully able to catch this in pycharm looking at the check-mark in the upper right corner before you submit the code.

I would like you to address the PEP8 violations before merging.
EDIT: I have noticed that the acceptance criteria might not have been entirely clear regarding PEP8, so please don't close this pull request until **this thursday**, when Yngve, the other TA's and me have talked it over.

**Comment by danielambrosius** [comment]
Hey!
Unfortunately your hand-in still won't pass. PEP8 quality control has been informed about in [this youtube video](https://www.youtube.com/watch?v=AyINGuxFpnU), and  the task discritpion:

> Check that your code passes the “quality control” in PyCharm!

Be sure to read through the task description before handing in the code as it sucks to fail on formalities.

**Comment by Andkleven** [comment]
d

**Pull request merged**